### PR TITLE
extend the attributes object with helper methods

### PR DESCRIPTION
**Pull request by chris-rock**
- add helper methods
- allow the attribute object to generate ruby code

### CODE REVIEW

File: lib/inspec/objects/attribute.rb [hunk 1]
```diff
@@ -21,13 +21,34 @@ def default
...
+    def title
+      @opts[:title]
+    end
+
+    def description
+      @opts[:description]
+    end
+
+    def ruby_var_identifier
+      'attr_' + @name.downcase.strip.tr(' ', '-').gsub(/[^\w-]/, '')
```
Comment (alexpop) [inline lib/inspec/objects/attribute.rb]: PR looks good. What you think of updating this slug maker from:

``` ruby
[10] pry> " hello\tdear  spaces\n".downcase.strip.tr(' ', '-').gsub(/[^\w-]/, '')
=> "hellodear--spaces"
```

To:

``` ruby
[11] pry> " hello\tdear  spaces\n".downcase.strip.gsub(/\s+/, '-').gsub(/[^\w-]/, '')
=> "hello-dear-spaces"
```

Comment (chris-rock) [reply lib/inspec/objects/attribute.rb]: good point, lets use your approach

**Comment by alexpop** [comment]
Looks good, merging. Thanks Chris!

**Comment by aaronlippold** [comment]
Should this also have the `required` boolean as well as part of the object?

**Pull request merged**